src/net/udp_socket.cpp

```diff
@@ -0,0 +1,115 @@
+#include <net/udp_socket.hpp>
+#include <net/exception.hpp>
+
+namespace arch::net {
+UDPSocket::UDPSocket() :
+	Socket(Socket::protocol_t::UDP) {}
+UDPSocket::UDPSocket(port_type port) :
+	Socket(Socket::protocol_t::UDP, port) {}
+UDPSocket::UDPSocket(IPv4 address, port_type port) :
+	Socket(Socket::protocol_t::UDP, address, port) {}
+UDPSocket::~UDPSocket() {
+	Socket::~Socket();
+}
+
+bool UDPSocket::send_to(const Host& host, port_type port, const char* data, int len) {
+	sockaddr_in addr;
+	memset(&addr, 0, sizeof(addr));
+	addr.sin_addr = host.ip();
+	addr.sin_family = AF_INET;
+	addr.sin_port = htons(port);
+
+	int result = sendto(_socket, data, len, 0, (sockaddr*)&addr, sizeof(addr));
+	if (result == SOCKET_ERROR) {
+		throw NetException(gai_strerror(net_errno()));
+	}
+
+	return result;
+}
+bool UDPSocket::send_to(const Host& host, const char* data, int len) {
+	try {
+		return send_to(host, _port, data, len);
+	}
+	catch (NetException e) {
+		throw e;
+	}
+}
+bool UDPSocket::send_to(const Host& host, port_type port, std::string_view data) {
+	try {
```

Comment:
You don't have to rethrow errors. Whole sense of this system is that it's transparent to the caller. If you don't use try ... catch block here, the exception will be just propagated further in call stack.